### DISCUSSION TITLE
An error when deploying contract locally using forge

### DISCUSSION THREAD

**Discussion by mhmmadexsal** [Cyfrin/foundry-full-course-cu#1751] [General]
![Screenshot 2024-05-25 054123](https://github.com/Cyfrin/foundry-full-course-f23/assets/94267398/2e4479d6-af74-4fe8-8b3b-d389d1ef1bde)

// SPDX-License-Identifier: MIT
pragma solidity 0.8.18; // solidity version

contract SimpleStorage {
    // Declare favoriteNumber get initilized to 0 if no value is get 0

    uint256 myFavoriteNumber;

    struct Person {
        uint256 favoriteNumber;
        string name;
    }

    // Dynamic array
    Person[] public listOfPeople; /

    // mapping
    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        myFavoriteNumber = _favoriteNumber;
    }
    
    // view, pure
    function retrieve() public view returns (uint256){
        return myFavoriteNumber;
    }

    // memory, calldata, storage
    function addPerson (string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}

**Comment by EngrPips** [comment]
```js
// SPDX-License-Identifier: MIT
pragma solidity 0.8.18; // solidity version

contract SimpleStorage {
// Declare favoriteNumber get initilized to 0 if no value is get 0

uint256 myFavoriteNumber;

struct Person {
    uint256 favoriteNumber;
    string name;
}

// Dynamic array
Person[] public listOfPeople; /

// mapping
mapping(string => uint256) public nameToFavoriteNumber;

function store(uint256 _favoriteNumber) public {
    myFavoriteNumber = _favoriteNumber;
}

// view, pure
function retrieve() public view returns (uint256){
    return myFavoriteNumber;
}

// memory, calldata, storage
function addPerson (string memory _name, uint256 _favoriteNumber) public {
    listOfPeople.push(Person(_favoriteNumber, _name));
    nameToFavoriteNumber[_name] = _favoriteNumber;
}
}
```

**Reply by EngrPips** [reply]
Hello @mhmmadexsal, Are you deploying on sepolia chain or on anvil chain?

**Comment by urosognjenovic** [comment]
Make sure you're using one of the 10 available accounts created by Anvil when deploying a contract to a local chain.

I guess you used a private key from an external wallet (Metamask I guess) and it doesn't work because that account doesn't have any funds associated with Anvil.

**Comment by mhmmadexsal** [comment]
@thebraveman17 oke thank you so much

**Reply by urosognjenovic** [reply]
You're welcome. Please mark my reply as answer if it solved your problem.
